pkg/provisioner/conditionwaiter.go

```diff
@@ -52,13 +55,15 @@ func (w *conditionWaiter) Wait(ctx context.Context) (runtime.Object, error) {
 		}
 	}
 
+	workflow := azureutils.GetWorkflow(ctx, obj)
+
 	// if not wait for the event handler signal
 	select {
 	case <-ctx.Done():
-		klog.Errorf("context timeout for %s (%s)", w.objType, w.objName)
+		workflow.Logger().Error(status.Error(codes.DeadlineExceeded, "context timeout"), "context timeout")
 		return nil, ctx.Err()
 	case waitResult := <-w.entry.waitChan:
-		klog.Infof("received result for %s (%s): error :%v", w.objType, w.objName, waitResult.err)
+		workflow.Logger().Infof("received result for %s (%s): error :%v", w.objType, w.objName, waitResult.err)
```

Comment:
This should be a verbose log, if logged at all.

Reply:
I am still working on the verbosities. There still seem to remain many log lines to be removed.

Reply:
Consider making object type and name workflow (logger) details instead of including in the message string. We should also log errors as errors and omit a `nil` error in the success case.

Reply:
I removed the log portion and instead started a child workflow in `conditionWaiter` so that we can infer the results from the `workflow.Finish()`